### ISSUE TITLE
Zig build fails for root project

### ISSUE THREAD

**Issue by deevus** [godot-zig/godot-zig#39]
I noticed that if you try to run `zig build` for the root project it fails with the following message:

```
install
└─ install godot
   └─ zig build-lib godot Debug native 1 errors
src\api\Godot.zig:1:1: error: file exists in multiple modules
src\api\Godot.zig:1:1: note: root of module godot
src\api\Godot.zig:1:1: note: root of module root
src\api\Variant.zig:2:23: note: imported from module root
const Godot = @import("Godot.zig");
                      ^~~~~~~~~~~
error: the following command failed with 1 compilation errors:
C:\Users\Simon\scoop\persist\zvm\data\0.13.0\zig.exe build-lib -ODebug --dep godot --dep GodotCore --dep build_options -Mroot=C:\Users\Simon\Projects\godot-zig\src\api\Godot.zig -ODebug -I C:\Users\Simon\Projects\godot-zig\.zig-cache\o\9359d720b5f10679c03a8b8ac0c847ef\.zig-cache\tmp\dbadba7713a947da -I C:\Users\Simon\Projects\godot-zig\.zig-cache\tmp\a48f406bc2d4126a --dep GodotCore --dep build_options=build_options0 -Mgodot=C:\Users\Simon\Projects\godot-zig\src\api\Godot.zig -ODebug -I C:\Users\Simon\Projects\godot-zig\.zig-cache\tmp\a48f406bc2d4126a --dep godot -MGodotCore=C:\Users\Simon\Projects\godot-zig\.zig-cache\o\9359d720b5f10679c03a8b8ac0c847ef\.zig-cache\tmp\dbadba7713a947da\GodotCore.zig -Mbuild_options=C:\Users\Simon\Projects\godot-zig\.zig-cache\c\15052cdb85cbac4cb6d3d390991e1f18\options.zig -Mbuild_options0=C:\Users\Simon\Projects\godot-zig\.zig-cache\c\15052cdb85cbac4cb6d3d390991e1f18\options.zig -femit-implib --cache-dir C:\Users\Simon\Projects\godot-zig\.zig-cache --global-cache-dir C:\Users\Simon\AppData\Local\zig --name godot -dynamic --listen=-
Build Summary: 9/12 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install godot transitive failure
   └─ zig build-lib godot Debug native 1 errors
error: the following build command failed with exit code 1:
C:\Users\Simon\Projects\godot-zig\.zig-cache\o\9aa7499500dd251ef6d49b82be05e849\build.exe C:\Users\Simon\scoop\persist\zvm\data\0.13.0\zig.exe C:\Users\Simon\Projects\godot-zig C:\Users\Simon\Projects\godot-zig\.zig-cache C:\Users\Simon\AppData\Local\zig --seed 0x2cd2bd42 -Zc1abe64dfe316588
```

It doesn't appear to affect the example projects. 

Any thoughts @r-bar?

**Comment by deevus** [comment]
It is probably related to the circular dependency between Godot.zig and Variant.zig

**Comment by r-bar** [comment]
I will take a look at this tonight @deevus

**Comment by r-bar** [comment]
Ok, I have the root cause. When I built the original PR I got godot-zig to compile to a .so file first, then went back and added the module exports to make it able to be included from other zig projects. The issue is that means the root module `Godit.zig` actually gets built twice when you build from the base target. Once for the shared library (`lib.root_module`) and once in the exported zig module (`godot_module`) objects.

After I got the examples to compile I never went back and built the project from the root, assuming that it must still compile as well or the examples would not work. I apologize for that mistake. The build system is too smart for me and it trims the unnecessary shared library step when building the examples, because the examples only consume the zig modules.

The good news is that this was a known issue upstream and there was a [fix](https://github.com/ziglang/zig/pull/20388) for this exact scenario merged into zig master just last week.

The bad news is we don't have that fix available on 0.13.0 (or any other cut release). I would strongly prefer this library to remain buildable from an official zig release.

The rapid breaking build system changes means the build system is likely to break and need reworking for the next zig version. This build system will also likely never be able to support more than a single blessed zig release.

I think the best interim solution is to omit the shared library from the build since it serves no practical purpose. I will open a PR to that end.
